bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -741,6 +796,46 @@ private static boolean checkIsMapType(BType sourceType, BMapType targetType, Lis
         }
     }
 
+    private static boolean checkIsMapType(Object sourceVal, BType sourceType, BMapType targetType,
+                                          List<TypePair> unresolvedTypes) {
+        BType targetConstrainedType = targetType.getConstrainedType();
+        switch (sourceType.getTag()) {
+            case TypeTags.MAP_TAG:
+                return checkConstraints(((BMapType) sourceType).getConstrainedType(), targetConstrainedType,
+                                        unresolvedTypes);
+            case TypeTags.RECORD_TYPE_TAG:
```

Comment:
This case seems to have a big logic. shall we move it to a separate method for better readability?

Reply:
Done in 24985b7.